figanos/matplotlib/plot.py

```diff
@@ -1447,30 +1459,68 @@ def scattermap(
         "transform": transform,
         "zorder": 8,
         "marker": "o",
+        "edgecolor": "none",
     } | plot_kw
 
     plot_kw_pop = plot_kw.copy()
     for key in ["vmin", "vmax"]:
         plot_kw_pop.pop(key)
-
     # plot
-    ax.scatter(
-        x=plot_data.lon.values[mask],
-        y=plot_data.lat.values[mask],
-        c=plot_data.values[mask],
-        **plot_kw_pop,
-    )
+    plot_kw_pop = {"x": "lon", "y": "lat", "hue": plot_data.name} | plot_kw_pop
+    if ax:
+        plot_kw_pop.setdefault("ax", ax)
+    im = data.plot.scatter(**plot_kw_pop)
 
-    # colorbar
-    cbar_kw.setdefault("label", wrap_text(cbar_label))
-    cb = plt.colorbar(
-        mappable=matplotlib.cm.ScalarMappable(norm=norm, cmap=cmap), ax=ax, **cbar_kw
-    )
+    # add features
+    if ax:
+        ax = add_features_map(
+            data,
+            ax,
+            use_attrs,
+            projection,
+            features,
+            geometries_kw,
+            show_time,
+            frame,
+            plot_data,
+        )
+        if (frame is False) and (im.colorbar is not None):
+            im.colorbar.outline.set_visible(False)
+
+    else:
+        if sizes:
+            pt_sizes2 = np.resize(pt_sizes, sdata.shape)
+            n = 0
+        for fax in im.axes.flat:
+            fax = add_features_map(
+                data,
+                fax,
+                use_attrs,
+                projection,
+                features,
+                geometries_kw,
+                show_time,
+                frame,
+                plot_data,
+            )
+
+            # correct markersize for facetgrid
+            scat = fax.collections[0]
+            scat.set_sizes(pt_sizes2[n])
```

Comment:
I didn't think about 2D, I was trying to access sizes after plotting with Xarray since xr.plot.scatter really gives back weird result when it comes to the marker sizes even when given 's' in kwargs. Not sure that it is the best way to do that however (this is the part that's not working in scattter + facetgrids)

Reply:
wasn't working when I tested it befofe leaving, but seems to work now.